service/frontend/fx.go

```diff
@@ -275,26 +319,118 @@ func VisibilityManagerProvider(
 	)
 }
 
-func NamespaceReplicationQueueProvider(
-	serviceResource resource.Resource,
-) persistence.NamespaceReplicationQueue {
-	var namespaceReplicationQueue persistence.NamespaceReplicationQueue
-	clusterMetadata := serviceResource.GetClusterMetadata()
+func FEReplicatorNamespaceReplicationQueueProvider(
+	namespaceReplicationQueue persistence.NamespaceReplicationQueue,
+	clusterMetadata cluster.Metadata,
+) FEReplicatorNamespaceReplicationQueue {
+	var replicatorNamespaceReplicationQueue persistence.NamespaceReplicationQueue
 	if clusterMetadata.IsGlobalNamespaceEnabled() {
-		namespaceReplicationQueue = serviceResource.GetNamespaceReplicationQueue()
+		replicatorNamespaceReplicationQueue = namespaceReplicationQueue
 	}
-	return namespaceReplicationQueue
+	return replicatorNamespaceReplicationQueue
+}
+
+func ServiceResolverProvider(membershipMonitor membership.Monitor) (membership.ServiceResolver, error) {
+	return membershipMonitor.GetResolver(common.FrontendServiceName)
+}
+
+func AdminHandlerProvider(
```

Comment:
These exist mostly because we need to inject correct versions of dependencies. For example object can depend on two log.Logger and such a provider can hook them up correctly.

Reply:
Could you be more specific on what exactly does this give us? I see one to one mapping of all inputs from AdminHandlerProvider been passed to NewAdminHandler

Reply:
`	logger resource.SnTaggedLogger,
	throttledLogger resource.ThrottledLogger,`

Reply:
logger resource.SnTaggedLogger,